Composer/packages/client/src/pages/notifications/index.tsx

```diff
@@ -3,35 +3,41 @@
 
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import { useState } from 'react';
+import { useState, useContext } from 'react';
 import { RouteComponentProps } from '@reach/router';
-
+import { StoreContext } from '../../store';
 import { ToolBar } from './../../components/ToolBar/index';
 import useNotifications from './useNotifications';
 import { NotificationList } from './NotificationList';
 import { NotificationHeader } from './NotificationHeader';
 import { root } from './styles';
 import { INotification } from './types';
 import { navigateTo } from './../../utils';
-import { convertDialogDiagnosticToUrl } from './../../utils/navigation';
+import { convertDialogDiagnosticToUrl, convertInlineLgDiagnosticToUrl } from './../../utils/navigation';
 
-const navigations = {
-  lg: (item: INotification) => {
-    navigateTo(`/language-generation/${item.id}/edit#L=${item.diagnostic.range?.start.line || 0}`);
-  },
-  lu: (item: INotification) => {
-    navigateTo(`/dialogs/${item.id}`);
-  },
-  dialog: (item: INotification) => {
-    //path is like main.trigers[0].actions[0]
-    //uri = id?selected=triggers[0]&focused=triggers[0].actions[0]
-    const uri = convertDialogDiagnosticToUrl(item.diagnostic);
-    navigateTo(uri);
-  },
-};
 const Notifications: React.FC<RouteComponentProps> = () => {
   const [filter, setFilter] = useState('');
+  const { state } = useContext(StoreContext);
+  const { dialogs } = state;
   const notifications = useNotifications(filter);
+  const navigations = {
+    inlineLgTemplate: (item: INotification) => {
+      const url = convertInlineLgDiagnosticToUrl(item.id, dialogs);
+      navigateTo(url);
```

Comment:
![test12](https://user-images.githubusercontent.com/39758135/72327415-060e3e80-36ec-11ea-8102-6509993970c2.gif)

Reply:
if the link can not only navigate to the inline editor, but also can focus on the exactly `Line:Character` like all-up view has doing. it would be much better.

Reply:
we can navigate to the inline editor, but the convert function misses some conditions.

Reply:
bug fixed and we cannot locate at the specific line by url in the design page